google/cloud/bigquery/_helpers.py

```diff
@@ -430,6 +444,19 @@ def _record_field_to_json(fields, row_value):
         if subvalue is not None:
             record[subname] = _field_to_json(subfield, subvalue)
 
+        if isdict:
+            processed_fields.add(subname)
+
+    # Unknown fields should not be silently dropped.
+    if isdict:
+        not_processed = set(row_value.keys()) - processed_fields
+        if not_processed:
+            raise ValueError(
```

Comment:
Seems to me as though we should prefer keeping any unknown fields in the dictionary and let the API return an error.

Reply:
I was actually considering this, but then opted for more consistency with `tuple` rows, which also raise the error immediately if a length mismatch is detected.

On the other hand, though, we cannot detect all bad cases, and we definitely don't want to replicate the server-side logic in the client. I'll change it to include the unknown fields and let the backend complain.